nifi-nar-bundles/nifi-extension-utils/nifi-database-utils/src/test/java/org/apache/nifi/util/db/TestJdbcCommon.java

```diff
@@ -679,11 +680,12 @@ public void testConvertToAvroStreamForDateTimeAsLogicalType() throws SQLExceptio
 
         testConvertToAvroStreamForDateTime(options,
                 (record, date) -> {
+                    java.sql.Date expected = DataTypeUtils.convertDateToUTC(date);
                     final int daysSinceEpoch = (int) record.get("date");
                     final long millisSinceEpoch = TimeUnit.MILLISECONDS.convert(daysSinceEpoch, TimeUnit.DAYS);
-                    java.sql.Date actual = java.sql.Date.valueOf(Instant.ofEpochMilli(millisSinceEpoch).atZone(ZoneOffset.UTC).toLocalDate());
-                    LOGGER.debug("comparing dates, expecting '{}', actual '{}'", date, actual);
-                    assertEquals(date, actual);
+                    java.sql.Date actual = new java.sql.Date(millisSinceEpoch);
+                    LOGGER.debug("comparing dates, expecting '{}', actual '{}'", expected, actual);
+                    assertEquals(expected, actual);
```

Comment:
Instead of performing multiple conversions for the test, could this comparison be simplified by comparing `daysSinceEpoch` to the actual number of expected days?  That would make the assert much clearer.

Reply:
These tests (historically) do not assert the raw value (int / long) from the Avro record but reconstruct the Date/Time/Timestamp objects and compare them with the original objects (which were the input of the test case).
I believe the reason was to avoid to calculate the days/millis/etc. from the test input data which would be a similarly complicated method.

As the test inputs are literals, it is not needed to calculate the expected value from code, but it can be determined once and hard coded as literals too. What do you think?
I mean
```
int expectedDaysSinceEpoch = 17444
assertDate.accept(record, expectedDaysSinceEpoch);
```

Reply:
The code snippet you suggested looks much clearer and avoids the conversion, thanks!

Reply:
I could not go on that way because `testConvertToAvroStreamForDateTime()` is used in two places and the other really expects a Date object.
However, I found a simpler way to convert the Date to the expected number of days, and then to assert the raw value from Avro against that.